src/hooks/collaborators.ts

```
@@ -0,0 +1,121 @@
+  const [collaboratorsStatus, setCollaboratorsStatus] =
+    useState<QueryStatus>('pending');
+
+  const { data: accounts, status: accountsStatus } = useMy(
+    {
+      resource: 'account',
+    },
+    { doNotImpersonate: props?.doNotImpersonate }
+  );
+
+  const collaboratorEmails = useMemo(() => {
+    return accountsStatus === 'success'
+      ? accounts
+          .filter(
+            acc =>
+              !acc.key.endsWith('#settings#') && acc?.member === acc?.username
+          )
+          .map(acc => acc.name)
+      : [];
+  }, [JSON.stringify(accounts), accountsStatus]);
+
+  useEffect(() => {
+    const fetchCollaboratorsOrgName = async () => {
+      setCollaboratorsStatus('pending');
+
+      const requests = collaboratorEmails.map(async email => {
+        const promises = [];
+        promises.push(
+          axios({
+            method: 'get',
+            url: `/my?key=${encodeURIComponent('#account')}`,
+            headers: {
+              Authorization: token ? `Bearer ${token}` : '',
+              account: email,
+            },
+          }).then(response => response.data.accounts)
+        );
+
+        if (props?.getRiskCounts) {
+          promises.push(
+            axios({
+              method: 'get',
+              url: `/my/count?key=${encodeURIComponent('#risk')}`,
+              headers: {
+                Authorization: token ? `Bearer ${token}` : '',
+                account: email,
+              },
+            }).then(response => response.data)
+          );
+        }
+
+        const [accountResponse, risksResponse] = await Promise.all(promises);
+
+        const friendSettings = accountResponse.find((account: Account) =>
+          account.key.endsWith('#settings#')
+        );
+
+        return {
+          displayName: friendSettings?.config?.displayName || '',
+          email,
+          counts: risksResponse,
+        };
+      });
+
+      // Silence Error and display results that
```

Let's not silence errors but expose it to the user in some way, we shouldn't be getting errors under normal operating conditions. At least let's log this to `console.error` if there was an error. Most likely, we should be displaying snackbars if a request has failed.

---

Errors are silences as this is a bulk api, if this fails there could be multiple toast, so errors are silenced here and at the end if errors exist we make a error toast, by that instead of plenty of toast we make one toast


```
const results = await Promise.all(requests.map(p => p.catch(e => e)));
const hasSomeError = results.find(r => r instanceof Error);

if (hasSomeError) {
        Snackbar({
          variant: 'error',
          title: 'Failed to fetch some collaborators details',
          description: '',
        });
      }
```